tests/readiness_test.go

```diff
@@ -38,8 +36,42 @@ func TestReadiness(t *testing.T) {
 	clientset := runAgent(proxy.agent, stopCh)
 	waitForConnectedServerCount(t, 1, clientset)
 
+	// check the agent connected status
+	if !clientset.Ready() {
+		t.Fatalf("expected connection status 'true', got: :%t", clientset.Ready())
```

Comment:
Nit:

I would simplify verifidation to be similar to the server check above `t.Fatalf("expected not ready")`

The only possible failure value is "false", and the second call to clientsets.Ready() actually performs work (takes a lock, is subject to non-determinism bugs, etc). So if you prefer to keep it, please capture the value instead of calling twice.